src/charm.py

```diff
@@ -96,17 +103,9 @@ def _get_pebble_layer(self, jenkins_env: jenkins.Environment) -> ops.pebble.Laye
         }
         return ops.pebble.Layer(layer)
 
-    def _on_jenkins_pebble_ready(self, event: ops.PebbleReadyEvent) -> None:
-        """Configure and start Jenkins server.
-
-        Args:
-            event: The event fired when pebble is ready.
-        """
-        container = event.workload
-        if not container or not container.can_connect():
-            event.defer()
-            return
-
+    def _on_jenkins_pebble_ready(self, _: ops.PebbleReadyEvent) -> None:
+        """Configure and start Jenkins server."""
+        container = self.unit.get_container(self.state.jenkins_service_name)
```

Comment:
I'm wondering how this will affect events that are only fired once. Do we know if `pebble_ready` will be fired again if we don't do `event.defer()` ? I have the same question for `_on_storage_attached`

Reply:
Right, that is true, it seems like the precondition check may need to be called for every hook events as well. Let me write a spec for that as @jdkandersson has suggested. Turning this PR into draft for now. Thanks for the review!